src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -776,11 +776,19 @@ private void TrySetDirectType(FunctionMetadata metadata)
         // Ensure customer deployed application payload matches with the worker runtime configured for the function app and log a warning if not.
         // If a customer has "dotnet-isolated" worker runtime configured for the function app, and then they deploy an in-proc app payload, this will warn/error
         // If there is a mismatch, the method will return false, else true.
-        private static bool ValidateAndLogRuntimeMismatch(IEnumerable<FunctionMetadata> functionMetadata, string workerRuntime, IOptions<FunctionsHostingConfigOptions> hostingConfigOptions, ILogger logger)
+        private bool ValidateAndLogRuntimeMismatch(IEnumerable<FunctionMetadata> functionMetadata, string workerRuntime, IOptions<FunctionsHostingConfigOptions> hostingConfigOptions, ILogger logger)
         {
-            if (functionMetadata != null && functionMetadata.Any() && !Utility.ContainsAnyFunctionMatchingWorkerRuntime(functionMetadata, workerRuntime))
+            IEnumerable<FunctionMetadata> functionMetadataToCheck = functionMetadata ?? Enumerable.Empty<FunctionMetadata>();
+
+            // Placeholder mode will add "WarmUp" function of type CSharp. we should ignore this function for validating the metadata matching with worker runtime.
+            if (_environment.IsPlaceholderModeEnabled())
```

Comment:
Can't we simplify this and just skip the ValidateAndLogRuntimeMismatch call upstream entirely when in placeholder mode?

Reply:
Yes. The latest push simplified it. Thank you!

Reply:
@mathewc are you talking about performing a check where this code is called and skip the validation if in placeholder mode?

I like the safety of not requiring that check upstream, so future changes are less likely to re-introduce this problem.

Reply:
I'm good with the simpler approach, but to keep the best of both, can you perform a check in the method and throw if this is ever called in placeholder mode?

Reply:
Sure - either that, or just no-op return from the method if in placeholder mode

Reply:
Added a check in the method and throwing an exception now, if called from placeholder mode code path.